docs/Controller Guide.md

```diff
@@ -0,0 +1,100 @@
+
+### Notes on Understanding the AMWA NMOS Specifications
+All of the specifications referenced below can be accessed from the [AMWA NMOS](https://specs.amwa.tv/nmos/) website.
+
+As well as reading the text of the specifications, a fuller understanding of the described API can be achieved by paying particular attention to the API section (for instance, the [IS-04 Query API](https://specs.amwa.tv/is-04/releases/v1.3.1/APIs/QueryAPI.html)).
+
+APIs can be investigated by clicking on the HTTP methods for each endpoint listed (click [here](https://specs.amwa.tv/is-04/releases/v1.3.1/APIs/QueryAPI.html#senders_get) for an example).
+This will open a window with a description of the Request and Response for that HTTP method.
+
+These, along with the provided JSON schemas and examples, often contain canonical definitions not described in the supporting text.
+
+Similarly be aware of the [NMOS Parameter Registers](https://specs.amwa.tv/nmos-parameter-registers/) as this contains constant definitions used by many of the specifications.
+
+### IS-04 Discovery and Registration
+> *Controllers use the **IS-04 Query API** to discover and obtain updates on the NMOS resources that are available on the network.*
+
+The IS-04 specification describes the mechanism for the discovery and registration of NMOS resources within a media network. It comprises three APIs: the Registration API, the Query API and the Node API. Only the Query API has relevance to Controllers, but a brief summary of each API is given below for context.
+
+The **IS-04 Registration API** is exposed by the Registry. When a Node first joins the network, it uses [DNS-SD](https://specs.amwa.tv/info-004/) to discover the URL of the Registration API. It then uses the Registration API to register information about itself and all its sub-resources (Devices, Senders, Receivers, Sources and Flows). Each Node then communicates with the Registration API with a regular heatbeat message to let the Registry know that it is still on the network. Nodes also use the Registration API to update the information about themselves in the Registry whenever something changes. 
+
+The **IS-04 Query API** is also exposed by the Registry. Controllers discover the URL of the Query API using DNS-SD and then obtain an up-to-date list of all registered Nodes and sub-resources (Devices, Senders, Receivers, Sources and Flows) so that they can indicate to the user the currently available resources on the network. Controllers also use the Query API to set up websockets subscriptions to resources, so that they obtain updates every time something changes in the Registry.
+
+The **IS-04 Node API** is exposed by Nodes. It is used by other Nodes to discover information about a Node directly when there is no Registry available on the network (in peer-to-peer mode).
```

Comment:
It may also be used by controllers in a P2P environment.
Maybe we should say It is used by other Nodes or Controllers?

Reply:
Agreed. I have added Controllers.